### PR TITLE
Generate xml report

### PR DESCRIPTION
**Pull request by lisandrasilva**
This PR addresses issue #329.
When running `kontrol prove` with the flag `--xml-test-report` the report is generated. For now, the report only contains the tests that were passed in the `kontrol prove`.

### CODE REVIEW

File: src/tests/integration/test_foundry_prove.py [hunk 1]
```diff
@@ -633,3 +634,38 @@ def test_deployment_summary(
         TEST_DATA_DIR / 'foundry' / 'src' / 'DeploymentStateCode.sol',
         update=update_expected_output,
     )
+
+
+def test_foundry_xml_report(
```
Comment (lisandrasilva) [inline src/tests/integration/test_foundry_prove.py:639]: @ehildenb thank you for your suggestion! I just added the test. 

> Let's add a test to `test_foundry_prove.py` which exercises the new functionality.
> 
> * Pick a list of 3-5 short-running proofs. Run them in a batch with `foundry_prove(...)`.
> * Generate the XML report.
> * Re-load the XML report with a junit-style reader, to make sure it's understandable as an XML test report.

**Comment by lisandrasilva** [comment]
This PR is dependent on evm-semantics [PR #2304](https://github.com/runtimeverification/evm-semantics/pull/2304).

File: src/tests/profiling/test_foundry_prove.py [hunk 2]
```diff
@@ -28,27 +29,42 @@
...
+
+    tree = Et.parse('kontrol_prove_report.xml')
+    testsuites = tree.getroot()
+    testsuite = testsuites.find('testsuite')
+    assert testsuite
+    assert testsuite.get('name', 'None') == 'SimpleTest'
+    assert testsuite.findall('testcase[@name="test_assert_true()"]')
+    failure = testsuite.findall('testcase[@name="test_assert_false()"]')
+    assert failure
+    assert failure[0].findall('failure')
```
Comment (ehildenb) [inline src/tests/profiling/test_foundry_prove.py:67]: Is this intentional or should be reverted? It seems taht the test is added already to the integration test?

Comment (lisandrasilva) [reply src/tests/profiling/test_foundry_prove.py]: It was not intentional. I just reverted it.

File: src/kontrol/__main__.py [hunk 3]
```diff
@@ -806,6 +808,13 @@ def _parse_test_version_tuple(value: str) -> tuple[str, int | None]:
         action='store_true',
         help='Flag used by Simbolik to initialise the state of a non test function as if it was a test function.',
     )
+    prove_args.add_argument(
+        '--xml-test-report',
+        dest='xml_test_report',
+        default=False,
+        action='store_true',
+        help='Generate a JUnit xml report',
```
Comment (palinatolmach) [inline src/kontrol/__main__.py:816]: One very minor spelling suggestion: 

```suggestion
        help='Generate a JUnit XML report',
```

File: src/tests/integration/test_foundry_prove.py (hunk 1 above)
Comment (palinatolmach) [reply src/tests/integration/test_foundry_prove.py:739]: The test looks great, thanks @lisandrasilva! But it looks like the legacy backend job timed out on the previous run: https://github.com/runtimeverification/kontrol/actions/runs/8138402331/job/22251131183?pr=377. I'd say we can skip `test_foundry_xml_report` on legacy backend runs, e.g., you can just add
```
    if no_use_booster:
        pytest.skip()
```
to the test (you'll also need to add `no_use_booster: bool,` to the function signature).

**Review by palinatolmach — APPROVED** [review]
Approved with a few minor comments.

**Pull request merged**